Assets/Scripts/SS3D/Core/Systems/Rounds/RoundSystem.cs

```diff
@@ -0,0 +1,211 @@
+            if (permissionSystem.GetUserPermission(userCkey) != requiredRole)
+            {
+                Debug.Log($"[{nameof(RoundSystem)}] - User {userCkey} doesn't have {requiredRole} permission");
+            }
+            else
+            {
+                ServerStartWarmup();   
+            }
+        }
+
+        /// <summary>
+        /// Server method to start the warmup
+        /// </summary>
+        [Server]
+        private void ServerStartWarmup()
+        {
+            if (!IsServer)
+            {
+                return;
+            }
+
+            // Starts the warmup
+            _currentTimerSeconds = _warmupTimerSeconds;
+            UpdateRoundState(RoundState.WarmingUp);
+            
+            _warmupCoroutine = StartCoroutine(TickWarmup());
+
+            WarmupStartedMessage warmupStartedMessage = new();
+            InstanceFinder.ServerManager.Broadcast(warmupStartedMessage);
+        }
+
+        [Server]
+        private void HandleStartRound()
+        {
+            // Only do SyncVar assignments, tick coroutine and the RPC on the server.
+            if (!IsServer)
+            {
+                return;
+            }
+
+            if (RoundRunning)
+            {
+                Debug.Log($"[{nameof(RoundSystem)}] - Can't start round as round is already running");
+                return;
+            }
+            
+            UpdateRoundState(RoundState.Starting);
+            
+            UpdateRoundState(RoundState.Running);
+            StopCoroutine(_warmupCoroutine);
+            _tickCoroutine = StartCoroutine(Tick());
+            
+            InstanceFinder.ServerManager.Broadcast(new RoundStartedMessage());
+        }
+
+        [Server]
+        private IEnumerator TickWarmup()
+        {
+            while (_currentTimerSeconds > 0)
+            {
+                UpdateClock(GetTimerSeconds());
+                Debug.Log($"[{nameof(RoundSystem)}] - Start timer: {_currentTimerSeconds}");
+                _currentTimerSeconds--;
+                yield return new WaitForSeconds(1);
+            }
+
+            HandleStartRound();
+        }
+
+        [Server]
+        private IEnumerator Tick()
+        {
+            if (!IsServer)
+            {
+                yield break;
+            }
+            
+            while (RoundRunning)
+            {
+                UpdateClock(GetTimerSeconds());
+                _currentTimerSeconds++;
+                yield return new WaitForSeconds(1);
+            }
+
+            Debug.Log($"[{nameof(RoundSystem)}] - Coroutine running while round is not active");
+        }
+
+        [Server]
+        private void UpdateClock(int time)
+        {
+            if (!IsServer)
+            {
+                return;
+            }
+            
+            RoundTickUpdatedMessage roundTickUpdatedMessage = new(time);
+            InstanceFinder.ServerManager.Broadcast(roundTickUpdatedMessage);
```

Comment:
If you send a message here, can we remove the timer syncvar?

Reply:
Then leave the syncvar and remove the broadcast? I don't see why both are needed.

Reply:
The event can still be useful for future stuff, as the project grow we can remove it if we notice it is not needed. but already the event makes sure I don't have to get the round information on Update

Reply:
I'm unhappy with duplicated netmessages, but it's a tiny impact. Fine for now.